create-astro updates (#12083)
withastro/astro: .changeset/old-actors-learn.md

---
'create-astro': minor
---

> Rework the experience of creating a new Astro project.

Reviewer: ```suggestion
Reworks the experience of creating a new Astro project using the `create astro` CLI command.
```

> - Update the list of templates to include Starlight and remove the minimal template.

Reviewer: ```suggestion
- Updates the list of templates to include Starlight and combines the "minimal" and "basics" templates into a new, refreshed "Basics" template to serve as the single, minimal Astro project starter.
```
Maybe something like this is fine here, and then we don't need the paragraph at the end?

> - Removed the TypeScript question. Astro is TypeScript-only, so this question was often misleading. The "Strict" preset is now the default, but it can still be changed manually in `tsconfig.json`. `astro check` is no longer automatically added to the build script.

Reviewer: ```suggestion
- Removes the TypeScript question. Astro is TypeScript-only, so this question was often misleading. The "Strict" preset is now the default, but it can still be changed manually in `tsconfig.json`. 
- `astro check` is no longer automatically added to the build script.
```
I know they're both TypeScript related, but it seemed like there were different, separate reasons for removing `astro check` so this probably deserves its own line? And, we could put an explanation of why if you wanted!

> - Added a new question allowing the user to choose an adapter for their project, if they already know where they want to deploy it.

Reviewer: ```suggestion
- Adds a new question allowing the user to choose an adapter for their project, if they already know where they want to deploy it.
```

> Additionally, to support the removal of the minimal template as an option, this PR reworks the basics template to be more minimal, moving most of its implementation into Astro directly. This makes the Basics template now a mix between the old Basics and Minimal templates.

Reviewer: ```suggestion
```
We can probably get away without needing another statement here if we choose to work something into the list item above!

Author: I agree!

---
'create-astro': minor
---

> Add a question to ask the user which adapter to add. Add an `--add` flag taking a list of integration.

Reviewer 2: Need to talk about TS changes as well

Reviewer 3: I think `--add` should be split to a separate PR to avoid blocking on this review. Sounds it like we can release that independently.
